control_panel_api/helm.py

```diff
@@ -3,28 +3,37 @@
 from django.conf import settings
 
 
-def init_user(username, email, fullname):
-    helm_upgrade(
-        f'init-user-{username}',
-        'mojanalytics/init-user',
-        '--set', f'NFSHostname={settings.NFS_HOSTNAME}',
-        '--set', f'Username={username}',
-        '--set', f'Email={email}',
-        '--set', f'Fullname={fullname}',
-    )
-
-
-def config_user(username):
-    helm_upgrade(
-        f'config-user-{username}',
-        'mojanalytics/config-user',
-        '--namespace', f'user-{username}',
-        '--set', f'Username={username}',
-    )
-
-
-def helm_upgrade(release, chart, *flags):
-    default_flags = ['--install', '--wait']
-    flags = list(flags) + default_flags
-    subprocess.run(
-        ['helm', 'upgrade', release, chart] + flags, check=True)
+class Helm(object):
+
+    def __init__(self):
+        self.enabled = settings.ENABLED['write_to_cluster']
+
+    def _helm_command(self, command, *args):
+        if self.enabled:
+            subprocess.run(['helm', command] + list(args), check=True)
+
+    def upgrade_release(self, release, chart, *args):
+        default_flags = ['--install', '--wait']
+        flags = list(args) + default_flags
+        self._helm_command('upgrade', release, chart, *flags)
+
+    def init_user(self, username, email, fullname):
+        self.upgrade_release(
+            f'init-user-{username}',
+            'mojanalytics/init-user',
+            '--set', f'NFSHostname={settings.NFS_HOSTNAME}',
```

Comment:
Now we have a client object, could we extract the django settings stuff and use dependency injection?

Reply:
Yep, good idea

Reply:
Hm, I'm not so sure about this one actually. It would be better if we could remove this NFS_HOSTNAME variable from the API settings altogether. I think I'll leave it for now.